File: 兴趣班入门百练讲义/NQ100/NQ054/NQ054_improved.py
def main():
    import sys
    input = sys.stdin.read().split()
    idx = 0
    
    m = int(input[idx])
    idx += 1
    n = int(input[idx])
    idx += 1
    
    if m <= 0 or n <= 0:
        return
    
    scores = []
    for _ in range(m):
        row = list(map(int, input[idx:idx+n]))
        scores.append(row)
        idx += n
    
    max_row, max_col = 1, 1
    max_abs_score = abs(scores[0][0])
    
    for i in range(m):
        for j in range(n):
            current_abs = abs(scores[i][j])
            if current_abs > max_abs_score:
                max_abs_score = current_abs
                max_row, max_col = i + 1, j + 1
    
    print(max_row, max_col, scores[max_row-1][max_col-1])

if __name__ == "__main__":
    main()

File: 兴趣班入门百练讲义/NQ100/NQ054/NQ054_improved.cpp
#include <iostream>
#include <vector>
#include <cstdlib>
using namespace std;

int main() {
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);
    
    int m, n;
    cin >> m >> n;
    
    if (m <= 0 || n <= 0) return 0;
    
    vector<vector<int>> scores(m, vector<int>(n));
    for (int i = 0; i < m; ++i)
        for (int j = 0; j < n; ++j)
            cin >> scores[i][j];
    
    int max_row = 1, max_col = 1;
    int max_abs_score = abs(scores[0][0]);
    
    for (int i = 0; i < m; ++i) {
        for (int j = 0; j < n; ++j) {
            int current_abs = abs(scores[i][j]);
            if (current_abs > max_abs_score) {
                max_abs_score = current_abs;
                max_row = i + 1;
                max_col = j + 1;
            }
        }
    }
    
    cout << max_row << " " << max_col << " " << scores[max_row - 1][max_col - 1] << "\n";
    return 0;
}